crmd/remote_lrmd_ra.c

```diff
@@ -536,9 +536,40 @@ remote_lrm_op_callback(lrmd_event_data_t * op)
         (ra_data->cur_cmd == NULL) &&
         (ra_data->active == TRUE)) {
 
-        crm_err("Unexpected disconnect on remote-node %s", lrm_state->node_name);
-        ra_data->recurring_cmds = fail_all_monitor_cmds(ra_data->recurring_cmds);
-        ra_data->cmds = fail_all_monitor_cmds(ra_data->cmds);
+        if (crmd_is_rsc_managed(lrm_state->node_name)) {
+            crm_err("Unexpected disconnect on remote-node %s", lrm_state->node_name);
+            ra_data->recurring_cmds = fail_all_monitor_cmds(ra_data->recurring_cmds);
+            ra_data->cmds = fail_all_monitor_cmds(ra_data->cmds);
+        } else {
+            crm_info("Disconnect on unmanaged remote-node %s", lrm_state->node_name);
+            /* Do roughly what a 'stop' on the remote-resource would do */
+            g_hash_table_remove_all(lrm_state->pending_ops);
+            ra_data->active = FALSE;
+            lrm_state_disconnect(lrm_state);
+            if (ra_data->cmds) {
+                g_list_free_full(ra_data->cmds, free_cmd);
+            }
+            if (ra_data->recurring_cmds) {
+                g_list_free_full(ra_data->recurring_cmds, free_cmd);
+            }
+            ra_data->cmds = NULL;
+            ra_data->recurring_cmds = NULL;
+            remote_node_down(lrm_state->node_name, DOWN_KEEP_LRM);
```

Comment:
We definitely don't want to remove it from the cache -- we want it to be in the cache, but with an appropriate state. I think your original code is right; if a cluster node in maintenance mode goes away, it is still considered lost, and that is roughly equivalent to what we have here.